### DISCUSSION TITLE
Bug Report (Critical/Need HotFix)

### DISCUSSION THREAD

**Discussion by Kirid2005** [FakeFishGames/Barotrauma#14802] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The hacker is breaking the game by exploiting vulnerabilities. According to one of the Barotrauma developers, "Jasontti ff," the person is using "Seems like he is causing the lag spikes through overflowing sound buffer." When he joins the server, there is a spam of characters in the console that causes server lag and subsequently crashes it. Evidence will be provided below. 

He also cannot be banned by IP because he connects using a Steam ID with an unknown version of the game. 

He can crash all servers, whether public or private, whether P2P is used or not, whether it's a beta version or not. He can access everywhere. A password doesn't help either.

![Снимок экрана (58)](https://github.com/user-attachments/assets/435f42da-ac0c-49df-a46b-3a100d1c6869)
[Main_2024-10-01_0721 (2) (2).txt](https://github.com/user-attachments/files/17330540/Main_2024-10-01_0721.2.2.txt)

### Reproduction steps

It depends on luck. He can do this whenever he wants. There is no way to reproduce it. Just start the server and wait.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
I sent it higher. Error has too long characters.
```

**Comment by Regalis11** [comment]
Thank you for the report! 

We are aware of this issue, and are working on a fix. I believe we know what's causing this console spam, but the other issues (such as the hacker seeming to be able to execute commands without proper permissions) still need further investigation. If you happen to have any relevant serve logs or any other information, it might help us get to the bottom of the issue faster. I'm sorry for the inconvenience!

**Comment by Regalis11** [comment]
There's now a partial fix available to the issue in the latest unstable build (v1.6.14.0).

**Comment by Kirid2005** [comment]
![428013025-38401295-2828-451b-a303-a6d7aa593df7](https://github.com/user-attachments/assets/bb65b468-27e5-4f33-8d4b-d550529be43c)
[DebugConsoleLog_Server_03282025_1155.2.txt](https://github.com/user-attachments/files/19531739/DebugConsoleLog_Server_03282025_1155.2.txt)
[DebugConsoleLog_Server_03282025_1154.txt](https://github.com/user-attachments/files/19531744/DebugConsoleLog_Server_03282025_1154.txt)
[DebugConsoleLog_Server_03282025_1154.2.txt](https://github.com/user-attachments/files/19531745/DebugConsoleLog_Server_03282025_1154.2.txt)
[DebugConsoleLog_Server_03282025_1148.txt](https://github.com/user-attachments/files/19531747/DebugConsoleLog_Server_03282025_1148.txt)
[DebugConsoleLog_Server_03282025_1147.txt](https://github.com/user-attachments/files/19531752/DebugConsoleLog_Server_03282025_1147.txt)
[DebugConsoleLog_Server_03282025_1146.txt](https://github.com/user-attachments/files/19531754/DebugConsoleLog_Server_03282025_1146.txt)
[DebugConsoleLog_Server_03282025_1155.txt](https://github.com/user-attachments/files/19531756/DebugConsoleLog_Server_03282025_1155.txt)
[_2025-03-28_1127.txt](https://github.com/user-attachments/files/19531757/_2025-03-28_1127.txt)
[_2025-03-28_1132.txt](https://github.com/user-attachments/files/19531758/_2025-03-28_1132.txt)
[_2025-03-28_1137.txt](https://github.com/user-attachments/files/19531759/_2025-03-28_1137.txt)
[consolehostinglog.txt](https://github.com/user-attachments/files/19531760/consolehostinglog.txt)
![428004181-f8b01035-8adb-49ef-949a-747a8561a08c](https://github.com/user-attachments/assets/3d046aba-ee59-4357-bf43-8419656cff34)
![428004323-000d2c7a-111c-4d54-8ff4-1ac0b42a56e4](https://github.com/user-attachments/assets/68456f64-8e84-4bb5-9f4f-0bf56be4c33c)
![428005833-e4df718d-5646-4cdc-a736-996fe6115853](https://github.com/user-attachments/assets/a559f232-4f10-479d-9e32-e35342b47c14)

**Reply by Regalis11** [reply]
Thank you for the extra information! I believe I figured out how they're doing this, and implemented a safeguard against this in our private repo. We're trying to get the fix tested and included in next week update still. Sorry for the trouble - I hope we'll be able to squash this exploit for good with this fix!

**Comment by Kirid2005** [comment]
He can still break the server, but now without spamming symbols in the chat. Instead, he spams in other ways, like repeated connection attempts, etc. I tried to gather all the information during his "raid on the server." By the way, he can keep the server completely down for more than a day.

**Comment by Kirid2005** [comment]
@Regalis11 Another issue popped up. You did fix the previous exploit, but now a new one has appeared. According to my friends, people are joining with a modified client (again, based on what they said, they can change their skin tone to fully purple or only specific parts of the body). The result is that they join and crash the server. The host gets kicked from their own server, and other connected players experience the server freezing and then also getting kicked. This is specifically being triggered by players, not just a random server crash.

**Reply by Kirid2005** [reply]
> @Regalis11 Another issue popped up. You did fix the previous exploit, but now a new one has appeared. According to my friends, people are joining with a modified client (again, based on what they said, they can change their skin tone to fully purple or only specific parts of the body). The result is that they join and crash the server. The host gets kicked from their own server, and other connected players experience the server freezing and then also getting kicked. This is specifically being triggered by players, not just a random server crash.



We are already 50 people, and I congratulate everyone on this number. The number of interested users is growing daily. Most of you came here to have fun, but there are also those who came because they had no choice left—when everyone starts using cheats, you have to adapt, as fighting is useless.

Some people have asked me: "Why do all this? Why ruin the game for others?"

My answer: I am dissatisfied with the developers' policy regarding the game, the meager updates, and their dismissive attitude towards the players. At the moment of release, the price of the game was raised by more than double, and what did you get? Two and a half worthless updates with content stolen from mods and a broken PvP mode.

The developers don't give a damn about you or the money you spent.

More and more people realize this every day. To achieve this, certain measures have to be taken.

It might seem that 50 people is not a lot, but this number is enough to change the entire mess.

Each of you, with your own hands and actions, is changing the current reality.

I recently discovered a new vulnerability that allows me to shut down absolutely every server in the game, even dedicated ones. With the press of a single button.

Once our numbers grow, I will share it with you.
Therefore, I urge you to continue inviting your acquaintances and others here to bring this problem to light.

Play, break, spread.

Deus Ex Machina



This is a post by Norman from their discord server.

**Reply by Regalis11** [reply]
Thank you for letting us know! There are some fixes coming up in the next update that I suspect might be related to the exploit he's using, but we'll see, and continue to patch up holes he's found. Do you by any chance have any crash reports?

**Comment by Regalis11** [comment]
This issue should be fixed by some of the additional anti-DoS measures we've added in the latest update. Although, there is still a high chance Norman has more tricks up his sleeve, so we'll keep this ticket open still and keep an eye out for any further exploits that might pop up.

**Reply by Kirid2005** [reply]
> This issue should be fixed by some of the additional anti-DoS measures we've added in the latest update. Although, there is still a high chance Norman has more tricks up his sleeve, so we'll keep this ticket open still and keep an eye out for any further exploits that might pop up.

We didn’t have to wait long — he released a new version of the exploit with code obfuscation. True, you can still see the folders where he doesn’t want hidden content to show, but he bypassed the new Anti-DoS system, and the trick with the disperser (Swatiska) hasn’t been fixed.

**Reply by Regalis11** [reply]
Yeah, I'm looking at the updated dll right now. It seems he didn't bypass the new Anti-DoS measure, this is something slightly different. Should be fairly easy to address though, hopefully we can patched this new issue out soon too.

**Reply by Podrubchansky** [reply]
Cheaters continue to crash servers.
Profile of cheater: https://steamcommunity.com/profiles/76561198785753832
